### PR TITLE
docs(api_gateway): new event handler for API Gateway and ALB

### PR DESCRIPTION
**Pull request by heitorlessa**
**Issue #, if available:**

## Description of changes:

**Checklist**

 

* [x] [Meet tenets criteria](https://awslabs.github.io/aws-lambda-powertools-python/#tenets)
* [x] Update tests
* [x] Update docs
* [x] PR title follows [conventional commit semantics](https://github.com/awslabs/aws-lambda-powertools-python/blob/376ec0a2ac0d2a40e0af5717bef42ff84ca0d1b9/.github/semantic.yml#L2)
* [x] Create Getting Started example
* [x] Create sample infrastructure for Proxy Event
* [x] Create Examples section for high level methods
* [x] Cover correlation ID in the getting started example
* [x] Cover CORS for prototyping and production
* [x] Mention CORS on errors like 404
* [x] Cover integration with Data classes
  	- [x] query strings
  	- [x] headers, Lambda context
* [x] Cover use of Response()
* [x] Cover binary responses
* [x] Cover GZIP responses
* [x] Cover Testing
* [x] Cover FAQ on when we recommend Chalice and fully fledged web frameworks
* [x] ALB specifics
  	- ~~[x] Create sample infrastructure~~ 
  	  - ALB has no support for CloudFormation for Lambda as target
  	- [x] Demonstrate how to use ProxyEvent

**Minor implementation changes**

cc @michaelbrewer 

* [x] Enum to match code style (PascalCase)
* [x] Keep `not found` only in message return for security reasons
* [x] Remove `Tuple` type of response to keep it simpler (Dict and Response only)
* [x] Create Example section (docstrings) for routes `app.get, app.post, app.delete, etc.` to improve DX on IDEs.

## Breaking change checklist

**RFC issue #**:

* [ ] Migration process documented
* [ ] Implement warnings (if it can live side by side)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: aws_lambda_powertools/event_handler/api_gateway.py [hunk 1]
```diff
@@ -247,27 +247,27 @@ def __init__(self, proxy_type: Enum = ProxyEventType.APIGatewayProxyEvent, cors:
...
         """Post route decorator with POST `method`"""
         return self.route(rule, "POST", cors, compress, cache_control)
 
-    def put(self, rule: str, cors: bool = False, compress: bool = False, cache_control: str = None):
+    def put(self, rule: str, cors: bool = True, compress: bool = False, cache_control: str = None):
         """Put route decorator with PUT `method`"""
         return self.route(rule, "PUT", cors, compress, cache_control)
 
-    def delete(self, rule: str, cors: bool = False, compress: bool = False, cache_control: str = None):
+    def delete(self, rule: str, cors: bool = True, compress: bool = False, cache_control: str = None):
```
Comment (michaelbrewer) [inline aws_lambda_powertools/event_handler/api_gateway.py:262]: @heitorlessa  i don't think we want it like this.

Comment (michaelbrewer) [reply aws_lambda_powertools/event_handler/api_gateway.py:262]: @heitorlessa - i think the best we can do is default it to `None`  i can make a PR for this.

**Comment by heitorlessa** [comment]
Merging it now @michaelbrewer but I'll wait for your other PR to fix `cors=None` piece and I'll release it tomorrow morning.

**Review by michaelbrewer — APPROVED** [review]
LGTM. I will look at adding support for cors=None as the default at the route level.

**Pull request merged**